File: ecole-frontend/src/app/note.ts
import { Etudiant } from 'src/app/etudiant';
import { Matiere } from 'src/app/matiere';

export class Note {
  id: number = 0;
  valeur: number = 0;
  etudiant: Etudiant = new Etudiant();
  matiere: Matiere = new Matiere();
  noteDate!: Date;

  constructor(){}
/*  constructor(id: number, valeur: number, etudiant: Etudiant, matiere: Matiere) {
    this.id = id;
    this.valeur = valeur;
//    this.etudiant = etudiant;
    this.matiere = matiere;
  }*/
}

File: gestion-ecole/src/main/java/ma/edu/gestionecole/gestionecole/entities/Note.java
package ma.edu.gestionecole.gestionecole.entities;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;
import org.springframework.format.annotation.DateTimeFormat;

import javax.persistence.*;
import javax.validation.constraints.DecimalMax;
import javax.validation.constraints.DecimalMin;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import java.util.Date;

@Entity
@Data @AllArgsConstructor @NoArgsConstructor @ToString
public class Note
{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @NotNull
    @DecimalMin(value = "0", inclusive = true)
    @DecimalMax(value = "20")
    private double valeur;

    @Temporal(TemporalType.DATE)
    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private Date dateNote;

    @ManyToOne
    @NotNull
    private Etudiant etudiant;

    @ManyToOne
    @NotNull
    private Matiere matiere;
}
